sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -205,30 +206,145 @@ class Analyzer(
       CleanupAliases)
   )
 
+  object ResolveRecursiveReferneces extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = {
+      val recursiveTables = plan.collect {
+        case rt @ RecursiveTable(name, _, _) if rt.anchorResolved => name -> rt
+      }.toMap
+
+      plan.resolveOperatorsUp {
+        case UnresolvedRecursiveReference(name) if recursiveTables.contains(name) =>
+          RecursiveReference(name, recursiveTables(name).output.map(_.newInstance()))
```

Comment:
I added `_.newInstance()` because it simplifies `FoldablePropagation`. This way we don't need to mess with attributes in it. Please see initial commit for details.

Reply:
I see, may you please add a comment here in order to explain the reason for this?